pkg/apis/kubeone/types.go

```diff
@@ -111,6 +111,30 @@ type ClusterNetworkConfig struct {
 	ServiceSubnet     string `json:"serviceSubnet"`
 	ServiceDomainName string `json:"serviceDomainName"`
 	NodePortRange     string `json:"nodePortRange"`
+	CNI               *CNI   `json:"cni,omitempty"`
+}
+
+// CNIProvider type
+type CNIProvider string
+
+// List of CNI Providers
+const (
+	// https://docs.projectcalico.org/v3.7/getting-started/kubernetes/installation/flannel
+	// This provider does not support encryption.
+	CNIProviderCanal = CNIProvider("canal")
```

Comment:
What is the reason for using `CNIProvider("canal")` instead of just `canal`? For `CloudProviderName` we used the second approach. Should we make that consistent?